### ISSUE TITLE
Can't export IDL/FORTRAN code from Sandbox/AIRS.2013.12.08.001.L2.RetStd.v6.0.7.0.G13345141819 design.

### ISSUE THREAD

**Issue by hyoklee** [HDFGroup/hpd-ws#37]
I don't know why I can't download codes.

**Comment by ajelenak-thg** [comment]
I get an error windows in the app with this message:
```
Exporting design as IDL code failed:
IDL does not support datatype: H5T_STD_I8LE
```
IDL does not have a datatype comparable to signed 8-bit integer so those designs cannot be exported as IDL source. I believe FORTRAN is the same. (@jreadey can you confirm this?)

**Comment by ajelenak-thg** [comment]
My bad, FORTRAN *has* 8-bit signed integer datatype. I will remove "IDL" from this issues title.

**Comment by hyoklee** [comment]
Wait, let me confirm with IDL guru @mdpiper how we can deal with signed 8-bit integer in IDL code generation. For example, why don't you convert it to 16 bit integer and document that IDL code generation will perform type conversion automatically?

**Comment by mdpiper** [comment]
I agree with @hyoklee that IDL would need to use its 16-bit signed type to represent values on `[-128,127]`. However, be careful with allowing IDL to downcast, since it'll wrap negative values; e.g.:
    
    IDL> a = -5 ; 16-bit signed int
    IDL> byte(a)
     251

Let me know if I can provide more info.

**Comment by hyoklee** [comment]
Thanks @mdpiper for joining the discussion and suggesting a valuable input.

**Comment by ajelenak-thg** [comment]
> However, be careful with allowing IDL to downcast, since it'll wrap negative values

This is why I do not like the option of silent conversion. IDL code is not the only export option and presumably those users who care about IDL should not use 8-bit signed integers in their designs.

**Comment by hyoklee** [comment]
Please add an option to support silent conversion in [IDL code generation server API] (http://hpd-ws.readthedocs.org/en/latest/api.html#generate-idl-source-code). I think HPD users deserve such option when code generation request failed. If it can be stored in a user profile on the server, that will be even better.

**Comment by ajelenak-thg** [comment]
I disagree. The purpose of code exports is to create HDF5 template files that **exactly** follow the design. If a design uses 8-bit signed integer datatype and the IDL code actually replaces it with something else then the templates created by that IDL code will differ from the design. The promise of interoperability broken.

**Comment by hyoklee** [comment]
I don't think so. The purpose of code generation is to produce data in whatever language user prefers.  Interoperability issue is tied to netCDF, not with codes. You can put it on a vote for advisory committee.

**Comment by ajelenak-thg** [comment]
> The purpose of code generation is to produce data in whatever language user prefers.

No, not produce data but a template file: a file without data.

> Interoperability issue is tied to netCDF, not with codes.

Did you see the warning a few comments above on what can happen with downcasting from some other datatype back to 8-bit signed integer? This is much more basic interoperability problem then what you mention.

> You can put it on a vote for advisory committee.

Sure.

**Comment by hyoklee** [comment]
No, generated code is for the basis of mass data production, beyond template file. Jeffrey Lee has mentioned it before. 

Yes, I saw the warning but why do you care? If server generates 16 bit dataset creation  IDL code from 8 bit signed design, the resulting final HDF5 product will contain 16 bit signed integer data with 16 bit signed type. Data product end users will not care the fact that the original design was 8 bit and thus users don't have to downcast it. If HPD sever provides example reader code as well in the future, it can simply  generate code that reads it as 16-bit number.
